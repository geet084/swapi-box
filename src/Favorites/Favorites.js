import React from 'react'
import './Favorites.scss'
import Icon from '../Icon/Icon'

const baseURL = "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/";

const Favorites = ({showFavorites, active, numFaves, newScroll}) => {
  let classes;
  active === 'faves' ? classes = 'active' : classes = 'inactive'
  
  return (
    <div className="favorites">
      <h2
        className="swapi"
        onClick={newScroll}
      >
        SWAPI BOX
      </h2>
      <Icon
        url={baseURL + 'R2D2-icon.png'}
        choice={active}
        getInfo={showFavorites}
        classes={classes}
        optText={numFaves + ' favorites'}
      />
    </div>
  )
}

export default Favorites;