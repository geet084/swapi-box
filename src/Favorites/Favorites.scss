.favorites {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 50%;
  padding-right: 35px;
}

.swapi {
  color: darkgoldenrod;
  background-color: white;
  font-size: 2rem;
  margin: 0 200px 4px 0;
  padding: 0 10px;
  border-radius: 15px;
}