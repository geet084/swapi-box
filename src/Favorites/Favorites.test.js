import React from 'react'
import { shallow } from 'enzyme'
import Favorites from './Favorites'

describe('Favorites', () => {
  let wrapper;
  const mockNewScroll = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Favorites newScroll={mockNewScroll}/>
    )
  })

  it('should properly render snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should register a click on the button', () => {
    wrapper.find('.swapi').simulate('click');
    expect(mockNewScroll).toHaveBeenCalled();
  })

})