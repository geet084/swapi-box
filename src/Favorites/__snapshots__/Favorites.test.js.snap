// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should properly render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    newScroll={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="swapi"
          onClick={[MockFunction]}
        >
          SWAPI BOX
        </h2>,
        <Icon
          classes="inactive"
          optText="undefined favorites"
          url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/R2D2-icon.png"
        />,
      ],
      "className": "favorites",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI BOX",
          "className": "swapi",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "SWAPI BOX",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "classes": "inactive",
          "getInfo": undefined,
          "optText": "undefined favorites",
          "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/R2D2-icon.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="swapi"
            onClick={[MockFunction]}
          >
            SWAPI BOX
          </h2>,
          <Icon
            classes="inactive"
            optText="undefined favorites"
            url="http://icons.iconarchive.com/icons/sensibleworld/starwars/64/R2D2-icon.png"
          />,
        ],
        "className": "favorites",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI BOX",
            "className": "swapi",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "SWAPI BOX",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "classes": "inactive",
            "getInfo": undefined,
            "optText": "undefined favorites",
            "url": "http://icons.iconarchive.com/icons/sensibleworld/starwars/64/R2D2-icon.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
